Yosemite/Yosemite/Stores/ProductTagStore.swift

```diff
@@ -130,32 +126,19 @@ private extension ProductTagStore {
     /// onCompletion will be called on the main thread!
     ///
     func upsertStoredProductTagsInBackground(_ readOnlyProductTags: [Networking.ProductTag],
-                                                   siteID: Int64,
-                                                   onCompletion: @escaping () -> Void) {
-        let derivedStorage = sharedDerivedStorage
-        derivedStorage.perform { [weak self] in
-            self?.upsertStoredProductTags(readOnlyProductTags, in: derivedStorage, siteID: siteID)
-        }
-
-        storageManager.saveDerivedType(derivedStorage: derivedStorage) {
-            DispatchQueue.main.async(execute: onCompletion)
-        }
-    }
-
-    /// Deletes any Storage.ProductTag  that is not associated to a product on the specified `siteID`
-    ///
-    func deleteUnusedStoredProductTags(siteID: Int64) {
-        let storage = storageManager.viewStorage
-        storage.deleteUnusedProductTags(siteID: siteID)
-        storage.saveIfNeeded()
+                                             siteID: Int64,
+                                             onCompletion: @escaping () -> Void) {
+        storageManager.performAndSave({ [weak self] storage in
+            self?.upsertStoredProductTags(readOnlyProductTags, in: storage, siteID: siteID)
+        }, completion: onCompletion, on: .main)
     }
 
     /// Deletes any Storage.ProductTag with the specified `siteID` and `productID`
```

Comment:
np, the function takes the `ids` of the product tags, and not `productID`, we should update the comment.

Reply:
Good catch! I updated the comment in 31008d341b40ecd3dafd29b89584edc328c035f9.